tests/container-nesting

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+set -ex
+
+# Install and setup LXD.
+install_lxd
+lxd init --auto
+
+# Install dependencies
+install_deps jq
+
+poolDriverList="${1:-dir btrfs lvm zfs}"
+
+IMAGE="${TEST_IMG:-ubuntu-minimal-daily:24.04}"
+LXD_SNAP_CHANNEL="${LXD_SNAP_CHANNEL:-latest/stable}"
+
+# waitNestedInstanceReady: waits for the instance running within another LXD
+# instance to become ready (processes count > 1).
+waitNestedInstanceReady() (
+  { set +x; } 2>/dev/null
+  local instName=$1
+  local nestedInstName=$2
+
+  # Wait for the instance to report more than one process.
+  maxWait=60
+  processes=0
+  for _ in $(seq "${maxWait}"); do
+      processes="$(lxc exec "${instName}" lxc info "${nestedInstName}" | awk '{if ($1 == "Processes:") print $2}')"
+      if [ "${processes:-0}" -ge "${MIN_PROC_COUNT:-2}" ]; then
+          return 0 # Success.
+      fi
+      sleep 1
+  done
+
+  echo "===> FAIL: Nested instance ${nestedInstName} (on instance ${instName}) not ready after ${maxWait}s"
+  return 1 # Failed.
+)
+
+checkNestedInstanceDNS() {
+    local instName=$1
+    local nestedInstName=$2
+
+    # Test DNS resolution.
+    for _ in $(seq 1 5); do
+        if lxc exec "${instName}" -- lxc exec "${nestedInstName}" -- getent hosts canonical.com >/dev/null 2>&1; then
+            echo "===> PASS: DNS resolution: ${nestedInstName} (on instance ${instName})"
+            return 0
+        fi
+        sleep 1
+    done
+
+    echo "===> FAIL: DNS resolution: ${nestedInstName} (on instance ${instName})"
+    return 1
+}
+
+cat << EOF | lxc profile create container-nesting
+config:
+  limits.cpu: 4
+  limits.memory: 4GiB
+  security.nesting: true
+EOF
+
+if hasNeededAPIExtension devlxd_images_vm; then
+    lxc profile set container-nesting security.devlxd.images=true
+fi
+
+for poolDriver in ${poolDriverList}; do
+    poolName="pool-${poolDriver}"
+    instName="c1"
+
+    # Create storage pool.
+    lxc storage create "${poolName}" "${poolDriver}"
+
+    # Launch a new container and install LXD.
+    lxc launch "${IMAGE}" "${instName}" --profile default --profile container-nesting --storage "${poolName}"
+    waitInstanceReady "${instName}"
+    sleep 10 # To ensure the snapd is ready.
+
+    lxc exec "${instName}" -- timeout 30 snap wait system seed.loaded
+    lxc exec "${instName}" -- snap install lxd --channel="${LXD_SNAP_CHANNEL}"
+    lxc exec "${instName}" -- lxd init --auto
+
+    # Configure the list of storage drivers to be tested in the nested container.
+    # Storage driver "dir" is always supported.
+    nestedPoolDriverList="dir"
+
+    if [ "${poolDriver}" = "zfs" ]; then
```

Comment:
This should be gated by `hasNeededAPIExtension zfs_delegate` as this is only available with 5.0+.